src/python/pants/backend/python/macros/python_artifact.py

```diff
@@ -104,6 +86,3 @@ def __eq__(self, other: Any) -> bool:
 
     def __hash__(self) -> int:
         return self._hash
-
-    def __str__(self) -> str:
```

Comment:
I ran all the tests, and manually built some dists, with this `__str__` method modified to raise an exception, thus verifying that it's not actually called, let alone does it matter what this stringifies to.